hack/storage-version-migration.sh

```diff
@@ -65,19 +65,23 @@ spec:
   ttlSecondsAfterFinished: 600
 EOF
 
-JOB_NAME=$(kubectl -n shipwright-build get job --selector app=storage-version-migration-shipwright -o jsonpath='{.items[0].metadata.name}')
 
-while [ "$(kubectl -n shipwright-build get job "${JOB_NAME}" -o json | jq -r '.status.completionTime // ""')" == "" ]; do
-    echo "[INFO] Storage version migraton job is still running"
-    sleep 10
-done
```

Comment:
This is probably where the bug is. Per the Job API documentation:

```
completionTime(Time):

Represents time when the job was completed. It is not guaranteed to be set in happens-before order across separate operations. It is represented in RFC3339 form and is in UTC. The completion time is set when the job finishes successfully, and only then. The value cannot be updated or removed. The value indicates the same or later point in time as the startTime field.
```

I suspect that in your case, the job is failing. This script doesn't see a completion time because the `Job` never succeeds, hence it just runs forever.

See https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/job-v1/#JobStatus